lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -814,20 +814,16 @@ private void sendQueuedMessages() {
 
 	private void failQueuedMessages(ErrorInfo reason) {
 		synchronized(this) {
-			for (int n=0; n<queuedMessages.size(); ) {
-				QueuedMessage queued = queuedMessages.get(n);
-				try {
-					if(queued.listener != null) {
-						queuedMessages.remove(n);
+			for (QueuedMessage queued: queuedMessages) {
+				if (queued.listener != null) {
+					try {
 						queued.listener.onError(reason);
+					} catch (Throwable t) {
+						Log.e(TAG, "failQueuedMessages(): Unexpected error calling listener", t);
 					}
-					else {
-						n++;
-					}
-				} catch (Throwable t) {
-					Log.e(TAG, "failQueuedMessages(): Unexpected error calling listener", t);
 				}
 			}
+			queuedMessages.clear();
```

Comment:
Yes, it can mess up messages and even crash the app. I can change it but then we can get into infinite loop if onError() adds message and then we will call onError() on the new message and it adds a message too and so on